src/nanoarrow/nanoarrow.h

```diff
@@ -835,6 +838,15 @@ static inline ArrowErrorCode ArrowArrayAppendString(struct ArrowArray* array,
 /// length of the child array(s) did not match the expected length.
 static inline ArrowErrorCode ArrowArrayFinishElement(struct ArrowArray* array);
 
+/// \brief Finish a union array element
+///
+/// Appends an element to the union type ids buffer and increments array->length.
+/// For sparse unions, up to one element is added to non type-id children. Returns
+/// EINVAL if the underlying storage type is not a union, if type_id is not valid,
+/// or if child sizes after appending are inconsistent.
+static inline ArrowErrorCode ArrowArrayFinishUnionElement(struct ArrowArray* array,
+                                                          int32_t type_id);
```

Comment:
int8_t type_id?

Reply:
So, I would say there's no 'standard type ID', there's just the convention of making type ID == child index. Since we take type ID here, there's nothing more to do (in terms of this signature).